scripts/semver.sh

```diff
@@ -0,0 +1,71 @@
+#!/bin/bash
+
+###
+# semantic version comparition using semver specification http://semver.org/
+# This bash script compares pre-releases alphabetically as well
+#
+# returns 1 when A greater than B
+# returns 0 when A equals B
+# returns -1 when A lower than B
+#
+# Usage
+# chmod +x semver.sh
```

Comment:
Why is `chmod`-ing required? Git repos keep the permissions, therefore _if you commit_ this file with `+x` (or even `a+x`), it will be executable when cloned and checked out on any machine.

Reply:
I'm nor sure why but I ran into some permission issues on TravisCI without explicitly adjusting file permissions.

Then I found this:
https://stackoverflow.com/questions/39666585/does-git-store-the-read-write-execute-permissions-for-files

It could be a safety precaution to adjust permissions when needed to avoid relying on Git entirely.